From d5eab5f17b943ff9830c72b0a8f47eeef0be3b96 Mon Sep 17 00:00:00 2001
From: Nickey Yang <nickey.yang@rock-chips.com>
Date: Fri, 3 Mar 2017 15:56:35 +0800
Subject: [PATCH] arm64: dts: rockchip: use hdmi-ddc for ddc bus in rk3399

Using the builtin I2C controller in dw_hdmi is better than using the
normal RK3399 I2C controller(I2C3).

Change-Id: I6a2de7221263e8564f7cca56ea5c52e1a133c138
Signed-off-by: Nickey Yang <nickey.yang@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-android.dtsi          | 8 --------
 .../boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts | 7 -------
 arch/arm64/boot/dts/rockchip/rk3399.dtsi                  | 2 ++
 3 files changed, 2 insertions(+), 15 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-android.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-android.dtsi
index b6338ff4203d..bfda5a3f7e27 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-android.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-android.dtsi
@@ -252,18 +252,10 @@
 	status = "okay";
 };
 
-&i2c3 {
-	status = "okay";
-	i2c-scl-rising-time-ns = <450>;
-	i2c-scl-falling-time-ns = <15>;
-};
-
 &hdmi {
 	#address-cells = <1>;
 	#size-cells = <0>;
 	#sound-dai-cells = <0>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&hdmi_i2c_xfer>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
index c4e97df32633..44ad11832682 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
@@ -202,7 +202,6 @@
 	#address-cells = <1>;
 	#size-cells = <0>;
 	#sound-dai-cells = <0>;
-	ddc-i2c-bus = <&i2c3>;
 	status = "okay";
 };
 
@@ -219,12 +218,6 @@
 	};
 };
 
-&i2c3 {
-	status = "okay";
-	i2c-scl-rising-time-ns = <450>;
-	i2c-scl-falling-time-ns = <15>;
-};
-
 &i2s2 {
 	#sound-dai-cells = <0>;
 	status = "okay";
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index e7ce3ca71ef3..ede485142847 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -1845,6 +1845,8 @@
 		reg = <0x0 0xff940000 0x0 0x20000>;
 		reg-io-width = <4>;
 		rockchip,grf = <&grf>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&hdmi_i2c_xfer>;
 		power-domains = <&power RK3399_PD_HDCP>;
 		interrupts = <GIC_SPI 23 IRQ_TYPE_LEVEL_HIGH 0>;
 		clocks = <&cru PCLK_HDMI_CTRL>, <&cru SCLK_HDMI_SFR>, <&cru PLL_VPLL>, <&cru PCLK_VIO_GRF>;
-- 
2.35.3

